src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -1151,6 +1145,11 @@ public async Task ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream(boo
                             await connection.SendResponseBodyAsync("hello world");
                             break;
                     }
+
+                    if (connection is Http3LoopbackConnection http3Connection)
+                    {
+                        await http3Connection.WaitForClientDisconnectAsync();
```

Comment:
Just wondering if it's the same kind of a workaround we would have to add for https://github.com/dotnet/runtime/issues/57779?

Reply:
I wonder if we should just add this to the HTTP3 version of AcceptConnectionAsync.

Reply:
Yes, it's the same workaround. 

I tried to add it into `AcceptConnectionAsync` to make it more generic/future proof, but it caused timeouts in cancellation tests, so I scoped it to this particular test.

Side note: I think that the reason was that they dispose client one level higher then the server. Thus, the client connection is never closed before server needs to finish. Leading me to a conclusion that adding some generic solution might not be possible or at least won't be easy :cry:

Reply:
Ha @geoffkizer I didn't notice your comment before posting mine. I tried that, it cause timeouts elsewhere.

Reply:
If it's only a handful of cancellation tests that failed, it might make more sense to update those tests to work with this than to update all the other tests to call WaitForClientDisconnectAsync.

Cancellation tests are tricky anyway; I'd rather be modifying them to be more robust than modifying a bunch of more straightforward tests...

Do you remember which cancellation tests failed?

Reply:
I'll have to retry. I think that the number of test instances was high but that it was just 2 or 3 test methods repeated for different inputs.
So do not merge on my behalf, I'll retry tomorrow and either try to fix those or report back how much of them it was.

Reply:
Another option would be to actually initiate server shutdown here by sending a GOAWAY, then call WaitForClientDisconnectAsync. This is probably better behavior, but more involved.

Reply:
I posted a PR to do the graceful shutdown in AcceptConnectionAsync -- https://github.com/dotnet/runtime/pull/58088. Feedback welcome.

We should discuss if we want to take that PR or not.